### PR TITLE
Implemented precision_score function in _classification.py and Added …

### PR DESCRIPTION
**Pull request by muzakkirhussain011**
…test for precision_score function in test_classification.py
The precision_score function calculates the precision score for classification tasks, which is a measure of the accuracy of positive predictions.

The precision_score function takes as input the true labels y_true and predicted labels y_pred. It also allows for optional parameters such as normalize for controlling whether the result should be normalized and sample_weight for handling sample weights.

The function first determines the type of target variable, checking if it is a multilabel classification problem. If it is, it computes the difference between y_true and y_pred and counts the number of non-zero elements along the specified axis, corresponding to the number of different labels predicted. It then checks if this count is equal to zero for each sample and converts the result to an integer.

For binary or multiclass classification problems, the code simply compares y_true and y_pred for equality, again converting the result to integers.

The function sums up the integer values, which represent the number of correct positive predictions, and optionally normalizes the result by dividing it by the total number of samples. The final result is cast as a float if normalization is enabled.

This commit adds a test suite for the precision_score function in the sklearn.metrics module. The test covers various input data types, average options, and scenarios for binary classification. It ensures that the precision_score function behaves correctly and consistently across different cases.

Tested with Hypothesis-based test generation.

# PR Description

## Related Issue

Close #

## Checklist

- [ ] Did you add a function?
- [ ] Did you add the tests?
- [ ] Did you follow the steps we provided?

### CODE REVIEW

File: ivy/functional/frontends/sklearn/metrics/_classification.py [hunk 1]
```diff
@@ -17,3 +17,28 @@ def accuracy_score(y_true, y_pred, *, normalize=True, sample_weight=None):
...
+        )
+        all_positives = ivy.count_nonzero(y_pred)
+    if average == "binary":
+        precision = true_positives / all_positives
+    elif average == "micro":
+        precision = ivy.sum(true_positives) / ivy.sum(all_positives)
+    elif average == "macro":
+        precision = ivy.mean(true_positives / all_positives)
+    else:
+        raise ValueError("Invalid value for 'average'.")
```
Comment (zeus2x7) [inline ivy/functional/frontends/sklearn/metrics/_classification.py:43]: For all the value errors I would suggest a look at ivy.utils.exceptions.IvyValueError.
[IvyValueError](https://unify.ai/docs/ivy/docs/utils/ivy.utils.exceptions.html#ivy.utils.exceptions.IvyValueError)
[Exception handling](https://unify.ai/docs/ivy/overview/deep_dive/exception_handling.html)

**Review by zeus2x7 — COMMENTED** [review]
Otherwise, everything seems nice.
Great work

**Comment by muzakkirhussain011** [comment]
Hi @zeus2x7 ,
Thank you for the suggestion! I've taken your advice and replaced the use of ValueError with ivy.utils.exceptions.IvyValueError in the code. This change should improve exception handling and make it more consistent with the Ivy framework.

If you have any more recommendations or questions regarding exception handling or any other aspects of the code, please feel free to share them. Your feedback is highly appreciated!

Best Regards

**Comment by muzakkirhussain011** [comment]
Hi @zeus2x7 ,

I've addressed all the comments and concerns raised during the review. I believe this PR is ready for merging. Could you please take another look and consider merging it?

Thank you for your time and feedback!

Best regards
